### ISSUE TITLE
Signature Parse error when using @Transactional and Java 17

### ISSUE THREAD

**Issue by robelcik** [payara/Payara#5507]
# Description #
----------
_CDI deployment failure:Signature Parse error: Expected Field Type Signature_ exception is thrown when using `javax.transaction.Transactional` in Java 17 

## Expected Outcome ##

CDI beans with `@Transactional` annotations should work in Java 17

## Current Outcome ##

Exception is thrown when deploying. For example:
```
[2021-11-26T14:07:38.503+0100] [] [SEVERE] [NCLS-CORE-00026] [javax.enterprise.system.core] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1637932058503] [levelValue: 1000] [[
  Exception during lifecycle processing
org.glassfish.deployment.common.DeploymentException: CDI deployment failure:Signature Parse error: Expected Field Type Signature
        Remaining input: javax.transaction.Transactional$TxType -- Signature Parse error: Expected Field Type Signature
        Remaining input: javax.transaction.Transactional$TxType
        at java.base/sun.reflect.generics.parser.SignatureParser.error(SignatureParser.java:124)
        at java.base/sun.reflect.generics.parser.SignatureParser.parseFieldTypeSignature(SignatureParser.java:291)
        at java.base/sun.reflect.generics.parser.SignatureParser.parseFieldTypeSignature(SignatureParser.java:277)
```

## Steps to reproduce ## 

Deploy a .war with a bean like

```java
package com.example;

import javax.enterprise.context.ApplicationScoped;
import javax.transaction.Transactional;

@ApplicationScoped
public class Experiment {
    @Transactional
    void doSomething() {
        System.out.println("Hello!");
    }
}
```

## Environment ##

- **Distribution**: Server Full Profile, 5.2021.9
- **JDK Version**:  OpenJDK 64-Bit Server VM Zulu17.30+15-CA (build 17.0.1+12-LTS, mixed mode, sharing)
- **Operating System**: Linux

**Comment by Enforcer67** [comment]
Have you found a solution? @robelcik

**Comment by kasinskim** [comment]
Have a look at the JDK17 release notes document:
https://www.oracle.com/java/technologies/javase/17-relnote-issues.html#JDK-8265591


> Recompiling the sources or otherwise regenerating the class file to conform to the proper format will resolve the issue.

**Comment by robelcik** [comment]
> Have you found a solution? @robelcik

@Enforcer67 : Nope, I switched to EJB.

**Comment by rdebusscher** [comment]
Hi,

This is solved for our upcoming release 5.2021.10: https://github.com/payara/Payara/pull/5494

Regards
Rudy

**Comment by robelcik** [comment]
Awesome. Thank you!
